service/src/test/java/com/commercetools/util/HttpRequestUtilTimeoutTest.java

```diff
@@ -1,24 +1,112 @@
 package com.commercetools.util;
 
+import com.sun.net.httpserver.HttpServer;
 import io.sphere.sdk.http.HttpStatusCode;
 import org.apache.http.HttpResponse;
-import org.junit.Ignore;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Random;
 
 import static com.commercetools.util.HttpRequestUtil.*;
 import static java.lang.String.format;
+import static java.lang.Thread.sleep;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatIOException;
 
-@Ignore("14.02.2017: slowwly has issues now, so temporary skip the test")
-// should be re-factored: https://github.com/commercetools/commercetools-payone-integration/issues/213
 public class HttpRequestUtilTimeoutTest {
     /**
-     * Url pattern to <a href="http://slowwly.robertomurray.co.uk/">slowwly</a> http mocking resource.
-     * <p>
-     * <b>%d</b> is a placeholder for response delay in msec.
+     * Url pattern to local mocking http service.
+     * <ol>
+     * <li><b>%d</b> is a placeholder for socket port number.</li>
+     * <li><b>%d</b> is a placeholder for response delay in msec.</li>
+     * </ol>
+     */
+    private static final String URL_PATTERN = "http://localhost:%d?delay=%d";
+
+    private static Logger LOGGER = LoggerFactory.getLogger(HttpRequestUtilTimeoutTest.class);
+
+    private int testRandomPort;
+
+    /**
+     * Mock http service instance, initiated separately for every test case.
+     *
+     * @see <a href="https://docs.oracle.com/javase/8/docs/jre/api/net/httpserver/spec/com/sun/net/httpserver/HttpServer.html">
+     * HttpServer</a>
+     */
+    private HttpServer server;
+
+
+    /**
+     * Create mock http service which responses after some <i>delay</i>, specified in the the GET query.
      */
-    private static final String SLOWWLY_URL_PATTERN = "http://slowwly.robertomurray.co.uk/delay/%d/url/http://google.com";
+    @Before
```

Comment:
Why not `@BeforeClass`?

Reply:
Because looks like the service is not completely idempotent and stateless, thus i have to make it clean and fresh for every test case.